### ISSUE TITLE
rconExec issue: PacketFormatException - Unknown packet with header 4 received.

### ISSUE THREAD

**Issue by bennett1066** [koraktor/steam-condenser#252]
I am writing a Java program to send commands to my Rust server. I successfully connected to the server and sent commands. However, the server carries out the commands but comes back with the PacketFormatException: Unknown packet with header 4 received.
Any help would be appreciated.

Thanks

**Comment by koraktor** [comment]
As far as I know Rust implements the Source query protocol. This does not necessarily mean it also implements Source’s RCON protocol.

Is there any documentation about Rust’s server protocols available?

**Comment by bennett1066** [comment]
In this thread 'http://facepunch.com/showthread.php?t=1344705'  it says they use exactly the same protocol as the source engines RCON. It also directs people to 'http://forums.steampowered.com/forums/showthread.php?t=2946226'.
After looking at many examples I managed to write a method which would create packets and send out commands. This method works but does not read anything the server sends back. I am quite interested in knowing how many players are on so I need to read what the server sends back.
 The commands from rconExec work but I think the errors come from reading what the server sends back.

**Comment by koraktor** [comment]
Do you have an example server that could be used for testing?

**Comment by koraktor** [comment]
Any update on this one? Is this still an issue?

**Comment by bennett1066** [comment]
I actually closed my rust server due to the high amount of exploits on the legacy version. The newer version of rust is not quite playable yet but I might open a new server if it improves.

Rust appears to using a modified version of the Source RCON Protocol which is returning a packet type 4 which is different from what is described on this page https://developer.valvesoftware.com/wiki/Source_RCON_Protocol. I am not the only person to notice this http://help.howproblemsolution.com/139605/rcon-protocol-ignored.

I did read somewhere about the new experimental rust using a slightly different version of the RCON protocol but cannot find the link to that information. I managed to write a program which would send commands to the server but I did not get any information back from the server. I could not use the steam condenser because it would send a command and carry out that command but the program would crash with the unknown packet error.

Date: Sun, 4 Jan 2015 00:56:42 -0800
